### DISCUSSION TITLE
Deprecation warning in .dict() method used in https://fastapi.tiangolo.com/tutorial/body/#use-the-model

### DISCUSSION THREAD

**Discussion by suspiciousRaccoon** [fastapi/fastapi#11161] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from pydantic import BaseModel


class Item(BaseModel):
    name: str
    description: str | None = None
    price: float
    tax: float | None = None

app = FastAPI()


@app.post("/items/")
async def create_item(item: Item):
    item_dict = item.dict()
    if item.tax:
        price_with_tax = item.price + item.tax
        item_dict.update({"price_with_tax": price_with_tax})
    return item_dict
```

### Description

This is the example code used at https://fastapi.tiangolo.com/tutorial/body/#use-the-model.

When pasted over into vscode intellisense warns that the `dict` method at line 17 is deprecated and `model_dump` should be used instead.

### Operating System

Windows

### FastAPI Version

0.109.2

### Pydantic Version

2.6.1

### Python Version

3.11.2

**Comment by Kludex** [comment, marked as answer]
Yes. The documentation is reflecting what works with Pydantic V1, and V2, even if on V2 is deprecated.
